Normalize and split inputPath with path package, omit prefix creation

GeorgySerga opened the pull request:
Hello. I noticed that there is a dependency on the `buildIgnorePrefix` helper function. This function builds prefix, to split it later and ultimately get the `ignoreDepth` back again. Also prefix was always built with hard coded `../` delimiter sequence, what might cause problems on Windows machines, so I started normalizing the provided path and splitting by `path.sep`.

There is no issue associated, lmk if I should create one.

P.S. Some part pf the diff is prettier changes, I didn't change the config so I think it might be that prettier didn't work with some of the previous commits.

HeroProtagonist commented:
@GeorgySerga Thanks for this! Can you add in some tests that would represent a valid windows path. Let's get this PR merged and then rebase your other one since it touches the same files

GeorgySerga commented:
@HeroProtagonist Hello, just the status for now. I'm not sure how to write Windows related tests and the reason is that `path.normalize` and `path.sep` in
```
const pathArray = path.normalize(pathInput).split(path.sep)
```
would resolve to OS specific values (which is ultimately the goal). In general afaik it's not recommended to import modules with Windows deliminators, so it might be not a valid (widespread) issue, but I think the approach above is more generalized and handled by node logic itself, other than by splitting manually by the string
```
const pathArray = path.split('../')
```
That being said I used sources like: https://gist.github.com/domenic/2790533, https://github.com/ehmicky/cross-platform-node-guide/blob/master/docs/3_filesystem/file_paths.md and official docs for [normalize](https://nodejs.org/api/path.html#path_path_normalize_path) and [sep](https://nodejs.org/api/path.html#path_path_sep).
Normalize part says that:

> When multiple, sequential path segment separation characters are found (e.g. / on POSIX and either \ or / on Windows), they are replaced by a single instance of the platform-specific path segment separator (/ on POSIX and \ on Windows).

^ so we divide by one (unified) separator if we normalize and for `.sep`:

> On Windows, both the forward slash (/) and backward slash (\) are accepted as path segment separators; however, the path methods only add backward slashes (\).

which probably means that Windows developers don't choose (are not encouraged to choose) Windows specific separators. Also splitting by `path.sep` is "the docs way".

Do you have an idea how this can be tested cross OS? I think that the fact that previous tests pass means that this approach is not worse and should break existing functionality, as `.split('../')` would not split Windows paths for sure (`'foo\\bar\\baz'` from the docs).

That being said I don't have windows machine, so I can't give this code a proper manual Windows test either.
